### DISCUSSION TITLE
How to make a REPL with defined command/subcommands?

### DISCUSSION THREAD

**Discussion by legacy-code** [bilal-fazlani/commanddotnet#460] [Q&A] [answered]
I really like what I see with _commanddotnet_!  This is, by far the best framework for .Net I’ve used.  Keep up the awesome work!

I’m coming from using Python and Cmd2, which allows an interactive Repl with the commands defined.  I.e. you can run a command, ask for help, or whatever without exiting back to the command-line (unless or until you quit). Now I need to do something similar on Windows, but in C#.  Just not quite sure how to do this, as it doesn’t seem to be a usual use-case.

I have the commands defined and they work great, but I would like to just loop to wait on the next command instead of exiting the app.  I’m hoping there is an easy way to do this!

Any help or nudges would be appreciated!

Thanks,
Mike

**Comment by drewburlingame** [comment, marked as answer]
Hi @legacy-code, thanks for the compliments.  

This is something we have a poc for, but I've since had a child and this feature has fallen in priority for me.  I think all that remains are tests, but the tests for this are more complicated.  

Having said that, I did [publish a package](https://www.nuget.org/packages/CommandDotNet/6.1.0-alpha.repl) for it that some of used successfully.  

The PR is at https://github.com/bilal-fazlani/commanddotnet/pull/402.  That PR description and the example are the only docs for it.  You can let me know if you have questions.

Also, if you want to write some tests, I'd be happy to except PRs to get this pushed into main branch.

Cheers,
Drew

**Reply by legacy-code** [reply]
Hey @drewburlingame!  Congratulations!  As a parent of 5, I certainly know how that goes with time/priority!

Cool on the package.  I’ll give it a look and see what I can figure out!  This is a side project me a tools that I’m writing for myself, so I might not have too much time, but I will most likely have some questions for you at some point!  If I can get to it, I’ll most happily try to help out on tests!  I’ve found that testing helps me (read forces me) to understand thing better!

Thanks for getting back to me!  Really appreciate it!

Have a great one!
Mike
